### ISSUE TITLE
[ Microsoft.DotNet.ILCompiler ] "COM Interop is not supported on this platform" exception after updating to new version of package

### ISSUE THREAD

**Issue by FoggyFinder** [dotnet/runtimelab#399]
The same issue as a previous one but now with MCVE. Well, not entirely but at least it's a runnable project.

Didn't change anything except

```xml
<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="1.0.0-alpha-*" />
```
->
```xml
<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="6.0.0-*" />
```

StackTrace

```
Unhandled Exception: System.PlatformNotSupportedException: COM Interop is not supported on this platform.
   at Internal.Runtime.CompilerHelpers.InteropHelpers.ConvertManagedComInterfaceToNative(Object) + 0x42
   at Avalonia.Win32.Interop.UnmanagedMethods.RegisterDragDrop(IntPtr, IDropTarget) + 0x46
   at Avalonia.Win32.OleContext.RegisterDragDrop(IPlatformHandle, IDropTarget) + 0x68
   at Avalonia.Win32.WindowImpl.CreateDropTarget() + 0x43
   at Avalonia.Controls.TopLevel..ctor(ITopLevelImpl, IAvaloniaDependencyResolver) + 0x1ac
   at Avalonia.Controls.WindowBase..ctor(IWindowBaseImpl, IAvaloniaDependencyResolver) + 0x24
   at Avalonia.Controls.Window..ctor(IWindowImpl) + 0x62
   at Views.MainWindow..ctor() + 0x35
   at App.main@14-1.Invoke(Unit) + 0x1c
   at Gjallarhorn.Avalonia.SingleView`5.Run(FSharpFunc`2) + 0x41
   at Gjallarhorn.Avalonia.SingleView`5.Gjallarhorn-Bindable-Framework-INavigator`3-Run(ApplicationCore`3, FSharpFunc`2) + 0xe
   at Gjallarhorn.Avalonia.App.toApplicationSpecification@36.Invoke(FSharpFunc`2) + 0x26
   at App.main(String[]) + 0xd3
   at CoreRtAvaloniaMCVE!<BaseAddress>+0x10d217f
```

Repo with MCVE:

[CoreRtAvaloniaMCVE](https://github.com/FoggyFinder/CoreRtAvaloniaMCVE)

**Comment by jkotas** [comment]
Duplicate of #306 

May also get addressed by https://github.com/AvaloniaUI/Avalonia/pull/4971. cc @kekekeks @jkoritzinsky

**Comment by kekekeks** [comment]
It's planned to replace all of Avalonia's COM interop with microcom, but I can't guarantee that it will be in 0.10.0 release, probably in 0.10.1-0.10.5 timeframe

**Comment by FoggyFinder** [comment]
> Duplicate of #306

@jkotas  I doubt. Why then it works with CoreRt-1-*?

**Comment by jkotas** [comment]
COM interop has not been supported by CoreRT so far.

I suspect that it worked for you because of some second order effect, e.g. old version of Avalonia that did not have DragAndDrop support.

**Comment by FoggyFinder** [comment]
> I suspect that it worked for you because of some second order effect, e.g. old version of Avalonia that did not have DragAndDrop support.

No, it's not the case. The only line that was changed is

```xml
<PackageReference Include="Microsoft.DotNet.ILCompiler" Version="6.0.0-*" />
```

nothing more. You can check it by yourself if you run the sample.

**Comment by FoggyFinder** [comment]
Diff for dependencies:

https://www.diffchecker.com/cWLTuNsl

I see nothing suspicious

**Comment by jkotas** [comment]
Thank you for providing a repro. This seems to be side-effect of bug fix #266 .

Before this fix, the apartment state was not set correctly and drap&drop registration was silently failing due to this early out: https://github.com/AvaloniaUI/Avalonia/blob/0f65fadfd10d4c4544670b8a63d9e7638c9704c1/src/Windows/Avalonia.Win32/OleContext.cs#L39

After this fix, the apartment state is set correctly and drag&drop registration is failing with this exception.

You can work around this by deleting `[<STAThread>]`  in your app.fs. It will make the apartment  unset like before the fix #266 and your app will work as before.

**Comment by FoggyFinder** [comment]
Yay. Thank you. Fantastic - it works.
